### ISSUE TITLE
:wheelchair: a11y(tab-component): Keyboard interaction of tabs

### ISSUE THREAD

**Issue by carolabes** [baloise/design-system#1425]
### Description of this issue

* Tabs cannot be accessed with the usual keys

<img width="1371" alt="tabs" src="https://github.com/baloise/design-system/assets/172396366/ccb02204-5168-4d8b-9dfe-cd1b2a848ac4">

### How to fix it

* Tab keys should be accessible with usual keys like TAB, Arrows and DELETE (see additional Informations)

### Additional Information

Level A
https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html

Keyboard interaction: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#keyboard_interaction

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hirsch88** [comment]
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#keyboard_interactions

**Comment by hirsch88** [comment]
@carolabes We need to discuss this in our weekly.

There are 3 cases for the tabs

1. Tabs with only links should work with tab and enter
2. Tabs without any link should work with arrow keys
3. But we also have a mixture like in the main nav bar, would do we there?

Moreover, we get the feeling that we are confusing the end user with multiple solutions

**Comment by carolabes** [comment]
Hi @hirsch88,
the technical implementation is important for people who cannot see the design. Based on the implementation, they can identify the element and know how to access it. 
By definition, the tab changes parts of the page (see tab role definition). Technically we have only one use case here. 

This means:

1. Tabs with links only like our inpage navigation should be implemented as nav with link list.
2. Tabs that change parts of the content such as the tab component or product slider should be implemented as tabs and thus be operable.
3. In main nav bar no tabs are necessary, because it is a navigation. Here buttons and links are enough. 

I understand your point that it can be confusing for the users, but still the technical implementation should be correct for the users who don't see the design. 

If there are any questions, please do not hesitate to contact me.

**Comment by carolabes** [comment]
Checked on Baloise.ch: Accessing with right&left arrow key not possible

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tab_role#keyboard_interactions

**Comment by hirsch88** [comment]
Tabs & Links use Arrows home and end but not selected it only focus?
Tabs only navigate with arrow keys?

**Comment by GinaBiondo** [comment]
Arrow keys still not working for tabs or am I missing anything here? :)

**Comment by hirsch88** [comment]
We cannot reproduce the case. Could it be that one of your items is a link?

**Comment by GinaBiondo** [comment]
When trying to use the arrow keys with the tabs in storybook it is also not working. Those are not linked

**Comment by hirsch88** [comment]
Add prop to overwrite the keyboard navigation to use arrow or either tabulator. default is as it is
prop should tell if it is a linkedList or tabs

**Comment by mmarinkov** [comment]
remaining parts of this story will be handled in a separate case [#1679](https://github.com/baloise/design-system/issues/1679)
